### PR TITLE
sci-libs/hdf5: added ROS3 flag

### PR DESCRIPTION
**Pull request by TheChymera**
Closes: https://bugs.gentoo.org/814530
Package-Manager: Portage-3.0.23, Repoman-3.0.3
Signed-off-by: Horea Christian <chr@chymera.eu>

### CODE REVIEW

File: sci-libs/hdf5/hdf5-1.12.1.ebuild [hunk 1]
```diff
@@ -29,6 +29,10 @@ RESTRICT="!test? ( test )"
 
 RDEPEND="
 	mpi? ( virtual/mpi[romio] )
+	ros3? (
+		net-misc/curl
+		dev-libs/openssl
```
Comment (jsmolic) [inline sci-libs/hdf5/hdf5-1.12.1.ebuild:34]: Does it link to curl and openssl? If yes, we need to add `:=`

File: sci-libs/hdf5/metadata.xml [hunk 2]
```diff
@@ -6,13 +6,14 @@
...
-  than 2 GB, and supports parallel I/O.
-</longdescription>
+    HDF5 is a file format and library for storing scientific data. HDF5
+    was designed and implemented to address the deficiencies of HDF4.x. It
+    has a more powerful and flexible data model, supports files larger
+    than 2 GB, and supports parallel I/O.
+  </longdescription>
   <use>
     <flag name="hl">Enable high level API (https://support.hdfgroup.org/HDF5/doc/HL/index.html)</flag>
+    <flag name="ros3">Enable support for read-only S3 access (https://portal.hdfgroup.org/display/HDF5/Configuration+and+Setup+for+HDF5+Read+Only+S3+VFD)</flag>
```
Comment (jsmolic) [inline sci-libs/hdf5/metadata.xml:16]: I can't access this link?  
https://portal.hdfgroup.org/display/HDF5/Configuration+and+Setup+for+HDF5+Read+Only+S3+VFD

File: sci-libs/hdf5/hdf5-1.12.1.ebuild [hunk 3]
```diff
@@ -17,7 +17,7 @@ SRC_URI="https://www.hdfgroup.org/ftp/HDF5/releases/${MAJOR_P}/${MY_P}/src/${MY_
 LICENSE="NCSA-HDF"
 SLOT="0/${PV%%_p*}"
 KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux"
-IUSE="cxx debug doc examples fortran +hl mpi szip test threads unsupported zlib"
+IUSE="cxx debug doc examples fortran +hl mpi ros3 szip test threads unsupported zlib"
```
Comment (jsmolic) [inline sci-libs/hdf5/hdf5-1.12.1.ebuild:20]: You're adding a new USE flag and changing RDEPEND, so we need to revbump (https://devmanual.gentoo.org/general-concepts/ebuild-revisions/index.html)

**Review by jsmolic — CHANGES_REQUESTED** [review]

File: sci-libs/hdf5/metadata.xml (hunk 2 above)
Comment (TheChymera) [reply sci-libs/hdf5/metadata.xml:16]: @jsmolic I tried again, and the link seems to be working...

File: sci-libs/hdf5/hdf5-1.12.1.ebuild (hunk 1 above)
Comment (TheChymera) [reply sci-libs/hdf5/hdf5-1.12.1.ebuild:34]: @jsmolic I am pretty sure it doesn't (some other packages which depend on hdf5[ros3] fail some tests in the network sandbox and the error is clearly ascribed to the libs and not to hdf5). But how would I know for sure?

File: sci-libs/hdf5/hdf5-1.12.1.ebuild (hunk 3 above)
Comment (TheChymera) [reply sci-libs/hdf5/hdf5-1.12.1.ebuild]: done :)

File: sci-libs/hdf5/metadata.xml (hunk 2 above)
Comment (jsmolic) [reply sci-libs/hdf5/metadata.xml:16]: Yep, works for me now as well

File: sci-libs/hdf5/hdf5-1.12.1.ebuild (hunk 1 above)
Comment (TheChymera) [reply sci-libs/hdf5/hdf5-1.12.1.ebuild]: @jsmolic as per the discussion just now both are linked but only openssl has multiple slots

**Pull request closed without merging**